### PR TITLE
Write test for `walk_files` to check if it accepts Path-like and str objects

### PR DESCRIPTION
**Pull request by algomaster99**
* [x] ❗ Have you followed the guidelines in the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) list?

* [x] 📖 Check this box if this PR **does not** require [documentation](https://dvc.org/doc) updates, or if it does **and** you have created a separate PR in [dvc.org](https://github.com/iterative/dvc.org) with such updates (or at least opened an issue about it in that repo). Please link below to your PR (or issue) in the [dvc.org](https://github.com/iterative/dvc.org) repo.

* [x] ❌ Have you checked DeepSource, CodeClimate, and other sanity checks below? We consider their findings recommendatory and don't expect everything to be addressed. Please review them carefully and fix those that actually improve code or fix bugs.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

Reference: #2137

### CODE REVIEW

File: tests/unit/utils/test_utils.py [hunk 1]
```diff
@@ -139,3 +140,15 @@ def pattern(path):
...
+def test_walk_files(repo_dir):
+    path_list = []
+    for path in walk_files(repo_dir.root_dir):
+        path_list.append(path)
+
+    path_info_list = []
+    for path_info in walk_files(PathInfo(repo_dir.root_dir)):
+        path_info_list.append(path_info)
+
+    assert path_list == path_info_list
```
Comment (efiop) [inline tests/unit/utils/test_utils.py]: You could simply 
```
def test_walk_files(tmp_dir):
    assert list(walk_files(".")) == list(walk_files(tmp_dir))
```
`tmp_dir` is a new fixture we've introduced recently.

Comment (algomaster99) [reply tests/unit/utils/test_utils.py]: @efiop how will convert a path into `Path-like` object then?

Comment (efiop) [reply tests/unit/utils/test_utils.py]: @algomaster99 `tmp_dir` is a a path-like object :slightly_smiling_face:

**Comment by efiop** [comment]
@algomaster99 Could you please adjust and rebase?

**Comment by algomaster99** [comment]
@efiop I have made the necessary changes. Really, sorry for the delay. :sweat_smile: .

**Comment by efiop** [comment]
Thanks @algomaster99 ! 🙏

**Pull request merged**